torch/csrc/distributed/rpc/py_rref.cpp

```diff
@@ -46,6 +47,46 @@ RRefForkData fromPyTuple(const py::tuple& pyTuple) {
 
   return RRefForkData(ownerId, rrefId, forkId, parent, typeStr);
 }
+
+c10::optional<TypePtr> typePyObjToTypePtr(const py::object& type_value) {
+  if (type_value.is(py::none())) {
+    return c10::nullopt;
+  }
+  c10::QualifiedName type_value_str = c10::QualifiedName(py::cast<std::string>(
+      py::module::import("torch.jit").attr("_qualified_name")(type_value)));
+  return jit::get_python_cu()->get_type(type_value_str);
+}
+
+TypePtr decidePyObjIValueType(
+    const py::object& value,
+    c10::optional<TypePtr> type_hint) {
+  jit::InferredType type_inferred = jit::tryToInferType(value);
+  if (type_inferred.success()) {
+    // If we could infer the type from the pyobject, we create
+    // the RRef with the IValue of that type.
+    return type_inferred.type();
+  }
+
+  // If the py::object contains a ScripModule, we enforce users
+  // to specify it's ModuleInterface type.
+  c10::optional<jit::script::Module> module = jit::script::as_module(value);
+  if (module.has_value()) {
+    TORCH_CHECK(
+        type_hint.has_value(),
+        "If the RRef being created contains a ScriptModule, user must provide it's ModuleInterface type.");
+    // TODO: Check is subtype if ModuleInterface.
+    TORCH_CHECK(
+        type_hint.value()->kind() == TypeKind::InterfaceType,
+        "The py::object is a Module, type hint must be a ModuleInterface");
+    // TODO: Check object is instance of ModuleInterface.
+    return type_hint.value();
+  }
+
+  // Otherwise it's a spure pyobject, create the RRef
```

Comment:
Just curious about the word spure. Looks like this is not english? Or do you mean pure?

Reply:
Good catch. I think it should be "pure pyobject".